src/models/Integration.php

```diff
@@ -0,0 +1,379 @@
+<?hh // strict
+
+use Facebook\GraphNodes\GraphNode\Collection as GraphCollection;
+
+class Integration extends Model {
+
+  private function __construct(private string $type) {}
+
+  public function getType(): string {
+    return $this->type;
+  }
+
+  public static async function facebookOAuthEnabled(): Awaitable<bool> {
+    $oauth = Configuration::genFacebookOAuthSettingsExists();
+
+    if ($oauth) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  public static async function googleOAuthEnabled(): Awaitable<bool> {
+    $oauth = Configuration::genGoogleOAuthFileExists();
+
+    if ($oauth) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  public static async function facebookLoginEnabled(): Awaitable<bool> {
+    $login_facebook = await Configuration::gen('login_facebook');
+    $oauth = Configuration::genFacebookOAuthSettingsExists();
+
+    $login_facebook_enabled =
+      $login_facebook->getValue() === '1' ? true : false;
+
+    if (($oauth) && ($login_facebook_enabled)) {
```

Comment:
shouldn't generally add unneccessary brackets like this; the typechecker takes it as a "i know what I'm doing" hint in some situations, e.g. when assigning in an if condition

Reply:
We'll fix that here.  I wonder where else we may be using extra brackets within the platform.  We should take a look at this soon.